### ISSUE TITLE
Add informative error message when FigureWidget not available due to missing dependencies

### ISSUE THREAD

**Issue by vlizanae** [plotly/plotly.py#1111]
I'm running an alpine python3 container and for some reason FigureWidget is not available. Jupyter is not installed because it's meant to be a Dash app, are there any dependencies I might have overlooked?
![screenshot from 2018-08-14 17-56-03](https://user-images.githubusercontent.com/31910455/44118202-ba01f8c6-9feb-11e8-8f70-547b4c572d75.png)

**Comment by jonmmease** [comment]
Hi @vlizanae ,

The `FigureWidget` class won't be available unless `ipywidgets` is installed. If you're making a Dash app, then you'll want to use the `graph_objs.Figure` class instead of `FigureWidget`.  `FigureWidget` is essentially a subclass of `Figure` with extra logic and dependencies for integrating with the Jupyter Notebook.

Thanks for reporting this because it reminds me that it would be nice to do something more user friendly when someone tries to access a `FigureWidget` when the dependencies aren't available.

Alright if I change the title of this issue to "Raise informative error if FigureWidget dependencies are not installed"?

**Comment by vlizanae** [comment]
sure, thank you!

**Comment by Juanlu001** [comment]
I was thinking about this a bit and I don't see a quick & straightforward solution. This is the way `FigureWidget` is being imported in the `graph_objs` namespace:

```python
try:
    import ipywidgets
    from distutils.version import LooseVersion
    if LooseVersion(ipywidgets.__version__) >= LooseVersion('7.0.0'):
        from ._figurewidget import FigureWidget
    del LooseVersion
    del ipywidgets
except ImportError:
    pass
```

The only options I see:

* Keep this code, but instead of silencing the `ImportError`, display a warning. The problem is that lots of users that are not interested in `FigureWidget`s will see it.
* Do not import it in `graph_objs`, and instead make the user do `from plotly.graph_objs.widgets import FigureWidget`. This is an API breaking change, and also annoying to have this object in a different module.
* Add `ipywidgets>7` to the plotly.py requirements and get rid of the conditional import altogether.

Other ideas?

**Comment by nicolaskruchten** [comment]
Closed by #2445
